troncos/traces/decorate.py

```diff
@@ -161,6 +173,9 @@ def m3(self):
     """
 
     def _dec(cls: Type[TClass]) -> Type[TClass]:
+        if resource:
+            attributes["resource"] = resource
```

Comment:
Attributes can be None. Just pass `resource` into `_trace_function` below and be done with it.